In [1]:
%ls

 02_00_data_wrangling.ipynb
'A3_2 (1).ipynb'
'A3_Pivot Tables (1) (1).ipynb'
'A3_Pivot Tables (1).ipynb'
'A6_A00571517 (1).ipynb'
'A6_A00571517 (Error_Database).ipynb'
 A6_A00571517.ipynb
'A6_A00571517_Midterm Presentation.ipynb'
 adult.csv
 airline_data.csv
 antropometria00.csv
 antropometria-dataset.csv
 API_NY.GDP.MKTP.KD.ZG_DS2_en_csv_v2_3628907.csv
 Book3.csv
 Canada.xlsx
 Cartwheeldata.csv
 census.csv
 Class1
'Class2 (2).ipynb'
 Class2.ipynb
'Class_3 (1).ipynb'
'Class 4.ipynb'
'Class 5 (3).ipynb'
 coches.csv
 cwurData.csv
 data.csv
 datasets/
 headbrain.csv
 Metadata_Country_API_NY.GDP.MKTP.KD.ZG_DS2_en_csv_v2_3628907.csv
 Metadata_Indicator_API_NY.GDP.MKTP.KD.ZG_DS2_en_csv_v2_3628907.csv
 mic2.xlsx
 mic.xlsx
 movies.csv
 nhanes_2015_2016.csv
 pokemon.csv
 README.md
 reg_data.csv
 requirements.txt
 Reto/
 sample_pivot.csv
 tips.csv
 Untitled1.ipynb
 Untitled.ipynb
 weight-height.csv
 WorldBank.csv


In [2]:
import pandas as pd

In [6]:
df = pd.read_csv('reg_data.csv')
#df

In [7]:
pip install matplotlib

Note: you may need to restart the kernel to use updated packages.


In [8]:
pip install seaborn

     |████████████████████████████████| 292 kB 5.0 MB/s            
Note: you may need to restart the kernel to use updated packages.


In [10]:
pip install statsmodels

     |████████████████████████████████| 9.8 MB 6.2 MB/s            
     |████████████████████████████████| 233 kB 54.2 MB/s            
Note: you may need to restart the kernel to use updated packages.


In [11]:
pip install sklearn

  Preparing metadata (setup.py) ... done
     |████████████████████████████████| 24.8 MB 5.9 MB/s            
     |████████████████████████████████| 306 kB 58.0 MB/s            
  Created wheel for sklearn: filename=sklearn-0.0-py2.py3-none-any.whl size=1310 sha256=3c18c69a4ca0f39cd0ddb0b3f2fde1d90018374f433a6c71274bb5495b08e580
  Stored in directory: /home/jovyan/.cache/pip/wheels/46/ef/c3/157e41f5ee1372d1be90b09f74f82b10e391eaacca8f22d33e
Successfully built sklearn
Note: you may need to restart the kernel to use updated packages.


In [13]:
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import numpy as np
from sklearn import linear_model

In [26]:
df.dropna(inplace = True)
df

Gender  Salary   Age       Place  Weight Company Academic_degree
0   Female  1500.0  33.0     Chicago    80.0     BMW        Bachelor
1   Female  1200.0  33.0     Chicago    82.5    Ford              No
2     Male  2200.0  34.0    New York   100.8     BMW        Bachelor
3     Male  2100.0  42.0    New York    90.0     BMW          Master
4   Female  1500.0  29.0     Chicago    67.0    Ford          Master
5   Female  1700.0  42.0  Washington    60.0    Ford          Master
6     Male  3000.0  50.0  Washington    77.0    Ford              No
7     Male  3000.0  55.0  Washington    77.0    Ford        Bachelor
8   Female  2800.0  31.0    New York    87.0    Ford        Bachelor
9     Male  2900.0  46.0    New York    70.0      GM          Master
10  Female  2780.0  36.0  Washington    57.0     BMW              No
11    Male  2550.0  48.0    New York    64.0      GM          Master

1. Is there any effect of age on Salary? Use regression model to answer the question.

salary = b0 + b1*age

y = c + mx

y = df['Salary']
x = df['Age']

In [23]:
y = df['Salary']
x = df['Age']

type(x)

pandas.core.series.Series

In [28]:
y = pd.DataFrame(df['Salary'])
x = pd.DataFrame(df['Age'])

In [29]:
lm = linear_model.LinearRegression()

model_lm = lm.fit(x,y)
model_lm

LinearRegression()

In [31]:
print('The y intercept, b0 is = ', model_lm.intercept_)
print('The coefficient, b1 is = ', model_lm.coef_)
print('The R^2 is = ', model_lm.score(x,y))

The y intercept, b0 is =  [349.44580104]
The coefficient, b1 is =  [[48.09321584]]
The R^2 is =  0.380484541757693


# Salary = 349.44580104 + 48.09321584 * Age

Salary does not only depend on age.

In [ ]:
modelA = sm.OLS.from_formula('Salary ~ Age', data = df)
